### ISSUE TITLE
Uncertainty error code not consistent with its definition?

### ISSUE THREAD

**Issue by JoyceYa** [asharakeh/pod_compare#5]
Hi @asharakeh , thanks for releasing your code! Congratulations on the great work.

I have a problem with the calculation of uncertainty error in your code.

For example, when computing regression minimum uncertainty error, you use 
`reg_u_errors = 0.5 * ((sorted_gt_idxs_tp.sum(0) - tp_cum_sum) /
                                  sorted_gt_idxs_tp.sum(0)) + 0.5 * (fp_cum_sum / sorted_gt_idxs_fp.sum(0))`
the first numerator is not the number of TPs whose uncertainties are larger than an uncertainty threshold, and the second numerator is not the number of FPs whose uncertainties are smaller than the uncertainty threshold, besides, I did not find the uncertainty thresholds you set to get the uncertainty error.

Am I missing something, or you defined the uncertainty error differently here?

**Comment by asharakeh** [comment]
Hi @JoyceYa. So the equation provided in many papers is for the Uncertainty Error. What this code calculates is the **Minimum** uncertainty error. You can do this calculation in 2 ways:

1. Define millions of thresholds, compute the uncertainty error, find the minimum value.
2. Or you can compute a vector of every possible uncertainty error for your dataset without using any threshold by using the cumulative sum over ranked tp/fps from your dataset. This concept is very similar to how AP is calculated in BDD dataset evaluation code.

Basically, you can not use a single threshold, since you are computing the **minimum** uncertainty error, the lowest uncertainty error achieved by the detector over every single possible value of the threshold.

Hope this helps.

**Comment by JoyceYa** [comment]
You're right, thanks for the clear explanation!

By the way, you use gaussian differential entropy to represent regression uncertainty, is it possible the differential entropy has negative values ？ Will negative entropy values influence the evaluation?  How did you handle this problem in your implementation ?

**Comment by asharakeh** [comment]
Differential entropy for a gaussian has a support of [-inf, +inf]. Negative values should not impact the evaluation as no hardcoded thresholds are set for Uncertainty Error computations. Basically, negative values are also considered in the non-threshold-based UE computation.

However, if you want to use differential entropy in an application, things begin to get more complicated. In general, you should measure the differential entropy relative to some prior.
